### ISSUE TITLE
PDF scrolling is very laggy.

### ISSUE THREAD

**Issue by vanillaneutrino** [brave/brave-browser#13666]
## Description 

When viewing any PDF, the UI is very laggy/choppy.

## Steps to Reproduce

 Open a PDF and scroll. E.g. this PDF: https://www.parker.com/Literature/Aerospace%20Group/Aerospace%20Static%20Files/ParkerAerospace-4pgCapabilities.pdf

## Actual result:
Lag when scrolling

## Expected result:
Smooth scrolling, as reflected by the movement of the vertical scroll bar.

## Reproduces how often: 
For every PDF.

## Brave version (brave://version info)
Brave  1.19.86 Chromium: 88.0.4324.96 (Official Build) (64-bit)
Revision | 68dba2d8a0b149a1d3afac56fa74648032bcf46b-refs/branch-heads/4324@{#1784}
OS | Windows 10 OS Version 1809 (Build 17763.1697)

## Other Additional Information:

The issue is not reproducible in chrome.

**Comment by nihaalnz** [comment]
Its really annoying. Looking forward a fix for this. Gmail has a smooth scrolling pdf viewer.

**Comment by Arecsu** [comment]
Same here. And scrolling in Youtube while a video is being played is laggy as well.

Windows 10 20H2. 
Brave 1.19.86 Chromium: 88.0.4324.96 (Official Build) (64-bit)

**Comment by DotDebian** [comment]
Same here
Running Archlinux, latest build

**Comment by CavalloneChen** [comment]
Would it be possible to directly integrate pdf.js? https://github.com/mozilla/pdf.js/

**Comment by nihaalnz** [comment]
@CavalloneChen Using the extension on brave seems to give _smoothness_ back.

**Comment by CavalloneChen** [comment]
> @CavalloneChen Using the extension on brave seems to give _smoothness_ back. 
Yeah, that's true. But the extension seems unmaintained, the last update dates back to Sep. 2019.

**Comment by nihaalnz** [comment]
@CavalloneChen I don't know, so far it has what it needs. The basic features. I wouldnt worry much unless I get some kind of bugs or errors.

**Comment by DotDebian** [comment]
> @CavalloneChen Using the extension on brave seems to give _smoothness_ back.

Can you please link the extension ?

**Comment by vanillaneutrino** [comment]
@nihaalnz @CavalloneChen  I don't know if this helps, but if you go to brave://flags/ and search for "PDF" and enable "PDF viewer update" (brave://flags/#pdf-viewer-update) it enables a much better PDF viewer with more features and it is butter smooth. Perhaps you could enable this flag by default?

**Comment by vanillaneutrino** [comment]
@nihaalnz @CavalloneChen  I don't know if this helps, but if you go to brave://flags/ and search for "PDF" and enable "PDF viewer update" (brave://flags/#pdf-viewer-update) it enables a much better PDF viewer with more features and it is butter smooth. Perhaps you could enable this flag by default? (sorry I accidentally closed the issue)

**Comment by CavalloneChen** [comment]
> @nihaalnz @CavalloneChen I don't know if this helps, but if you go to brave://flags/ and search for "PDF" and enable "PDF viewer update" (brave://flags/#pdf-viewer-update) it enables a much better PDF viewer with more features and it is butter smooth. Perhaps you could enable this flag by default? (sorry I accidentally closed the issue)

Thanks for the advice, it is indeed much better.

**Comment by nihaalnz** [comment]
> @nihaalnz @CavalloneChen I don't know if this helps, but if you go to brave://flags/ and search for "PDF" and enable "PDF viewer update" (brave://flags/#pdf-viewer-update) it enables a much better PDF viewer with more features and it is butter smooth. Perhaps you could enable this flag by default? (sorry I accidentally closed the issue)

That is indeed useful, but for some reason, it still at times remains choppy, otherwise perfect, I think ill stick with the extension one.

**Comment by MungaiVic** [comment]
I'm also running Brave v89.1.22.71 on Ubuntu 20.10.
For PDFs stored locally, the response is good. For PDFs online, it is indeed choppy.

**Comment by cesss** [comment]
I'm not sure if this is the same issue, but the preview of attached PDFs in the web version of Outlook takes several minutes to appear (at least on MacOS using an old i3). I installed PDF.js from the Chrome store, and then the PDF previews are instantly shown when in the Outlook web version. So, the builtin PDF viewer in Brave doesn't seem to be Outlook-friendly. The older PDF.js works better.

**Comment by Poopooracoocoo** [comment]
This issue is especially noticeable when using a touchpad or touchscreen.
PDF.js was added and then removed in favour of PDFium: #3846
Crbugs to watch/star: [1316936](https://bugs.chromium.org/p/chromium/issues/detail?id=1316936), [1312420](https://bugs.chromium.org/p/chromium/issues/detail?id=1312420), [1306236](https://bugs.chromium.org/p/chromium/issues/detail?id=1306236), [1304913](https://bugs.chromium.org/p/chromium/issues/detail?id=1304913), [1285196](https://bugs.chromium.org/p/chromium/issues/detail?id=1285196), [1284144](https://bugs.chromium.org/p/chromium/issues/detail?id=1284144), [1278476](https://bugs.chromium.org/p/chromium/issues/detail?id=1278476), [1217012](https://bugs.chromium.org/p/chromium/issues/detail?id=1217012), [1187577](https://bugs.chromium.org/p/chromium/issues/detail?id=1187577), [1093611](https://bugs.chromium.org/p/chromium/issues/detail?id=1093611), [781689](https://bugs.chromium.org/p/chromium/issues/detail?id=781689), [775938](https://bugs.chromium.org/p/chromium/issues/detail?id=775938)
An extension you can use to get PDF.js to render PDFs: https://add0n.com/pdf-reader.html
PDF.js has significantly better scrolling but comes at the cost of significantly increased memory usage.
There are specific crbugs linked above that report performance issues with PDFs that are image heavy such as the PDF in this issue.

**Comment by Sneezy123** [comment]
I have the same issue with Brave Version 1.91.172 (64-Bit), Chromium: 149.0.7827.115

I am using Fedora 43 KDE. The mentioned flag does not appear in my brave://flags.
